app/features/quizzify/core.py

```diff
@@ -13,11 +13,16 @@ def executor(files: list[ToolFile], topic: str, num_questions: int, verbose=Fals
 
         # Instantiate RAG pipeline with default values
         pipeline = RAGpipeline(verbose=verbose)
+        logger.debug(f"Before Compile")
         
         pipeline.compile()
+
+        logger.debug(f"Before Pipeline")
```

Comment:
While this technically works, it might not be the most effective use of the logger.debug method. The logger.debug method is typically used to provide detailed information that can help in diagnosing issues, usually including variable values, state information, or other data that might assist in understanding the flow of the code or pinpointing where something might be going wrong.

In the current usage, the message is very brief and does not provide much context. To make better use of logger.debug, we could include more specific information about the state of the application at this point in the code.